src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -56,7 +57,10 @@ public KustoClient(KustoDataProviderConfiguration config, string requestId)
 
         public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, string database, string requestId = null, string operationName = null)
         {
-
+            if(query!=null && query.Contains("Tenant in ()"))
```

Comment:
This logic would also catch cases where people have comments in the query.
Instead of this, can we let the query run and catch Kusto Bad Exception here and then check if the exception has Tenant in ()

Reply:
That will make sure that the query indeed failed due to missing Tenant Info

Reply:
Good point. This will be a stronger catch.

Reply:
Although even in that case a query can be a bad query due to other reasons and still have a commented "Tenant in ()".